packages/did/src/DidDetails/FullDidDetails.utils.ts

```diff
@@ -28,10 +28,32 @@ const methodMapping: Record<string, VerificationKeyRelationship | undefined> = {
   web3Names: 'authentication',
 }
 
-export function getKeyRelationshipForExtrinsic(
-  extrinsic: Extrinsic
+export function getKeyRelationshipForExtrinsic<A extends AnyTuple = AnyTuple>(
+  call: CallBase<A>
 ): VerificationKeyRelationship | undefined {
-  const { section, method } = extrinsic.method
+  const { section, method } = call
+
+  // get the VerificationKeyRelationship of a batched call
+  if (
+    section === 'utility' &&
+    (method === 'batch' || method === 'batchAll' || method === 'forceBatch') &&
+    call.args[0].toRawType() === 'Vec<Call>'
+  ) {
+    // map all calls to their VerificationKeyRelationship and deduplicate the items
+    const keys = new Set(
+      (call.args[0] as any as Array<CallBase<A>>).map(
+        (innerExtrinsic: CallBase<A>) => {
+          return getKeyRelationshipForExtrinsic(innerExtrinsic)
```

Comment:
A bit convoluted to get one key relationship and making sure, that the rest of the array does not contain another one.
I would suggest just getting the key relationship from the first item and than using `Array.every` (or `.some`, depending on which way you want to test) to test for uniqueness.

Reply:
Yes I guess I was a bit focused on my newly acquired `Set` skills. 😁 I would propose to use `reduce` since that's what we do, we reduce the array to a single value. I guess it's not as efficient as `every`/`some` since it's checking every value and doesn't short circuit, but I thought that might be irrelevant.

Reply:
yes, better than throwing on duplicate.